rust-arroyo/src/backends/kafka/producer.rs

```diff
@@ -118,9 +117,9 @@ impl ArroyoProducer<KafkaPayload> for KafkaProducer {
         // If the producer fails to flush the message out of the buffer, the delivery callback will send an error
         // on the channel.
         // match rx.recv_timeout(Duration::from_secs(5)) {
-        match rx.recv() {
+        match rx.recv_timeout(Duration::from_secs(5)) {
```

Comment:
for the record i think we should make the produce function here return a future, so to align it with python. but it's not that important and we can postpone it